### DISCUSSION TITLE
Is there anyway to have a single resolver resolve multiple fields?

### DISCUSSION THREAD

**Discussion by davlee1972** [mirumee/ariadne#864] [Q&A]
If there any way to resolve multiple fields with results from a single SQL call? Ideally I don't want to even call the SQL if none of the dependent fields need to be resolved.

Example in theory:

@user.field(["firstName","lastName"])
def resolve_firstName(obj, info):
    """" select first_name, last_name from people_table where user_id = obj["user_id"] """
    first_name = "Bob"
    last_name = "Marley"
    return first_name, last_name

**Comment by rafalp** [comment]
> Ideally I don't want to even call the SQL if none of the dependent fields need to be resolved.

What's your implementation? Normally the SQL query would occur in resolver for field returning the user, and not in field resolving user's attributes (unless those are separate database entities).

`firstName` resolver's job should be to get database data from parent resolver as `obj` and find user's first name in it, not query database for this name. Ergo, `obj` should be the user data, not parent object containing `user_id`.

**Reply by davlee1972** [reply]
For example

I have two sql tables

User id, first name, last name

User id, flight destination, travel date

If I create a resolver for flight history with user id as a search option for the resolver I can get their flight data.

However if the client also wants to include first name, last name or both then I need to make another sql call. 

I do not want to make a sql call.in the flight resolver unless the clients wants a name.

I also do not want to make two sql calls for the same record to resolve first name and last name independently.

**Reply by davlee1972** [reply]
I think my work around will be create separate resolvers for first name and last name, but try try pull either one from a dictionary cache first and populate the cache with a sql call if this the first attempt to resolve either name.

BTW I'm guessing it isn't possible to set other obj fields using a single resolver other than field being decorated

**Reply by fortyfree93** [reply]
So even for primitive Types a resolver is needed ?
Let's say I've got a schema like:
```
type type123 {
	field_01: Float!
	field_02: Float!
}
```

I got a resolver for ObjectType 'type123' in which I query the data from databaes and still have to run seperate resolvers for 'field_01' & 'field_02'. Or is there a way to avoid the field resolvers ?

**Reply by fortyfree93** [reply]
Ah sorry, I just realized that I simply have to use fallback_resolvers.

**Reply by davlee1972** [reply]
Your schema should include something like:
my_column: type123

with a resolver on my_column which returns a value of  {'field_01': ???, 'field_02': ???}
